pkg/util/id.go

```diff
@@ -16,31 +16,81 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
 	"io/ioutil"
+	"os"
+	"path"
 	"strings"
+
+	"github.com/google/uuid"
+)
+
+const (
+	maxNodeIDLength = 12
```

Comment:
with a full 32 character uuid it feels like we're being verbose in the etcd namespace. It's overkill if all we need is uniqueness in a cluster.